test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -14218,14 +14218,25 @@ def create_subjob_id(self, job_array_id, subjob_index):
         return job_array_id[:idx + 1] + str(subjob_index) + \
             job_array_id[idx + 1:]
 
-    def create_eatcpu_job(self, duration=None):
+    def create_eatcpu_job(self, duration=None, mom=None):
         """
         Create a job that eats cpu indefinitely or for the given
         duration of time
         """
+        if self.du is None:
+            self.du = DshUtils()
         script_dir = os.path.dirname(os.path.dirname(__file__))
         script_path = os.path.join(script_dir, 'utils', 'jobs', 'eatcpu.py')
-        DshUtils().chmod(path=script_path, mode=0o755)
+        pbs_conf = self.du.parse_pbs_config(mom)
+        shell_path = os.path.join(pbs_conf['PBS_EXEC'],
+                                  'bin', 'pbs_python')
+        a = {ATTR_S: shell_path}
+        self.set_attributes(a)
```

Comment:
keeping in mind that run_copy() can fail, I suggest moving this part after if condition